src/fe-common/core/fe-common-core.c

```diff
@@ -461,26 +461,36 @@ void fe_common_core_finish_init(void)
 
 gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 {
+	int channel_type = module_get_uniq_id_str("WINDOW ITEM TYPE", "CHANNEL");
+	int query_type = module_get_uniq_id_str("WINDOW ITEM TYPE", "QUERY");
+	char **tmp;
+	
 	g_return_val_if_fail(array != NULL, FALSE);
+	g_return_val_if_fail(dest != NULL, FALSE);
+	g_return_val_if_fail(dest->window != NULL, FALSE);
+	g_return_val_if_fail(dest->target != NULL, FALSE);
 
-	if (strarray_find(array, "*") != -1)
-		return TRUE;
+	WI_ITEM_REC *item = window_item_find_window(dest->window, dest->server, dest->target);
+	g_return_val_if_fail(item != NULL, FALSE);
 
-	if (strarray_find(array, dest->target) != -1)
-		return TRUE;
+	for (tmp = array; *tmp != NULL; tmp++) {
+		if (dest->server_tag != NULL && g_str_has_prefix(g_ascii_strdown(*tmp, -1), g_ascii_strdown(dest->server_tag, -1))) {
```

Comment:
Both g_ascii_strdown leak, consider g_ascii_strncasecmp() with the length parameter set to the short one